content/posts/raft-consensus-algorithms/index.md

```diff
@@ -0,0 +1,148 @@
++++
+date = "2022-04-01T10:00:00+07:00"
+author = "duchh"
+description = "Explaining how Raft work to achieve consensus between nodes"
+title = "Raft consensus algorithms"
+categories = ["Raft", "Consensus algorithm"]
+tags = ["Raft", "Consensus-algorithm"]
+slug = "raft-consensus-algorithms"
++++
+
+***In this blog post, we will learn about Raft and the way it work to achieve consensus between nodes.***
+
+#### **1. Consensus algorithm**
+
+Consensus algorithm means you have a cluster node, you want all of the nodes in the cluster to have the same information, data,... 
+In order to do that, Consensus applies a mechanism in the cluster we will have one server Leader, it has mission replicate data for 
+all of other servers in the cluster. Another thing Consensus algorithm guarantee available for the cluster, for example, when sever Leader
+is crashed the cluster will have another server instead immediately. Consensus algorithm properties include: safety, availability, integrity. 
+Consensus algorithm is usually used in distributed systems to solve some consensus issues.
+
+![alt](./images/master-slave.png)
+
+#### **2. Raft algorithm**
+Raft is algorithm for managing a replicated log of the form. Raft implements by first electing a server Leader, then giving the Leader complete 
+responsibility for managing the replicated log. The Leader accepts log entries from client, replicate them on other servers, and tell servers
+when it is safe to apply log entries to their state machine. 
+
+##### **2.1. Term**
+* **Term** is a number marked for the period at the beginning of Leader voting until the vote is successful, the new Leader will manage 
```

Comment:
'is a number marked for the period at the beginning of Leader voting until the vote is successful,'
Is difficult to undestand.